internal/pkg/policy/policy_output.go

```diff
@@ -245,6 +269,46 @@ func (p *Output) prepareElasticsearch(
 	return nil
 }
 
+func (p *Output) createRemoteEsClientIfNotExists(ctx context.Context, bulker bulk.Bulk, outputMap smap.Map) error {
+	remoteEs := bulker.GetRemoteClient(p.Name)
+	if remoteEs != nil {
+		// TODO replace if hosts or service token changed?
```

Comment:
I am wondering how can I decide if the `hosts` or `service_token` is different in the output config than the config used to create the esClient originally. I could store these values in the bulker too.

Does anyone know how does it work for the local ES, does the bulker pick up the hosts or other config changes and recreate the es client?

Reply:
Currently if fleet-server [receives a new config](https://github.com/elastic/fleet-server/blob/main/internal/pkg/server/fleet.go#L196) it will try to detect a [change it the subsystems it runs](https://github.com/elastic/fleet-server/blob/main/internal/pkg/server/fleet.go#L242). If a change is detected, the entire server is [stopped and restarted](https://github.com/elastic/fleet-server/blob/main/internal/pkg/server/fleet.go#L177-L190).

We may want to store the output config in each bulker (as a `map[string]interface{}`) and use `reflect.DeepEqual` to see if it has changed.
I don't think that storing the `hosts` and `service_token` is enough as the output can include additional information (proxy config, certs, etc)

Reply:
So do you think it makes sense to add all policy output configs to the [Config](https://github.com/elastic/fleet-server/blob/main/internal/pkg/config/config.go#L40) object and use it in [configChangedServer](https://github.com/elastic/fleet-server/blob/main/internal/pkg/server/fleet.go#L242) logic?

I'm not sure though what's the best way to do this. Currently the remote bulkers are in `policy_output.go` which is probably not the best place, I don't think I can call anything from `fleet.go` from here as it would be a circular dependency. Maybe a new shared channel between these files to send config changes, if that's possible?

Another solution could be to store the remove bulkers map in the main bulker, and create a channel in the main bulker to send config changes, to which we could listen to in `fleet.go`.

Reply:
Implemented the second approach in this [commit](https://github.com/elastic/fleet-server/pull/3051/commits/a52b32be3d15c8231bf9f92f917feaeb23344d6e)